MantidPlot/src/Mantid/InstrumentWidget/InstrumentWindow.cpp

```diff
@@ -1,837 +1,58 @@
- */
-void InstrumentWindow::saveSettings() {
-  QSettings settings;
-  settings.beginGroup("Mantid/InstrumentWindow");
-  if (m_InstrumentDisplay)
-    settings.setValue("BackgroundColor",
-                      m_InstrumentDisplay->currentBackgroundColor());
-  auto surface = getSurface();
-  if (surface) {
-    // if surface is null istrument view wasn't created and there is nothing to
-    // save
-    settings.setValue("PeakLabelPrecision",
-                      getSurface()->getPeakLabelPrecision());
-    settings.setValue("ShowPeakRows", getSurface()->getShowPeakRowsFlag());
-    settings.setValue("ShowPeakLabels", getSurface()->getShowPeakLabelsFlag());
-    settings.setValue("ShowPeakRelativeIntensities",
-                      getSurface()->getShowPeakRelativeIntensityFlag());
-    foreach (InstrumentWindowTab *tab, m_tabs) { tab->saveSettings(settings); }
-  }
-  settings.endGroup();
-
-  QSettings qs;
-  qs.beginGroup(QString::fromStdString(m_windowRendSettingsGroup));
-
-  // sets the combo to the particular value but does
-  // not update the setSurface? combo value
-  qs.setValue("user_params_Full3d", m_renderTab->m_full3D->isChecked());
-  qs.setValue("user_params_m_cylindricalX",
-              m_renderTab->m_cylindricalX->isChecked());
-  qs.setValue("user_params_m_cylindricalY",
-              m_renderTab->m_cylindricalY->isChecked());
-  qs.setValue("user_params_m_cylindricalZ",
-              m_renderTab->m_cylindricalZ->isChecked());
-  qs.setValue("user_params_m_sphericalX",
-              m_renderTab->m_sphericalX->isChecked());
-  qs.setValue("user_params_m_sphericalY",
-              m_renderTab->m_sphericalY->isChecked());
-  qs.setValue("user_params_m_sphericalZ",
-              m_renderTab->m_sphericalZ->isChecked());
-  qs.setValue("user_params_m_sideBySide",
-              m_renderTab->m_sideBySide->isChecked());
-  qs.setValue("user_param_mAxisCombo", m_renderTab->mAxisCombo->currentText());
-  qs.setValue("user_params_flipCheckBox",
-              m_renderTab->m_flipCheckBox->isChecked());
-
-  // saving the settings for the graph and color map
-  qs.setValue("user_param_minVal",
-              m_renderTab->m_colorMapWidget->getMinValue());
-  qs.setValue("user_param_maxVal",
-              m_renderTab->m_colorMapWidget->getMaxValue());
-  qs.setValue("user_param_nthTerm",
-              m_renderTab->m_colorMapWidget->getNth_power());
-  qs.setValue("user_params_autoscaling",
-              m_renderTab->m_autoscaling->isChecked());
-
-  // saving the bin range entered
-  qs.setValue("user_params_minBinRange", m_instrumentActor->minBinValue());
-  qs.setValue("user_params_maxBinRange", m_instrumentActor->maxBinValue());
-
-  qs.endGroup();
+std::string InstrumentWindow::saveToProject(ApplicationWindow *app) {
+  TSVSerialiser tsv;
+  tsv.writeRaw("<instrumentwindow>");
+  tsv.writeLine("WorkspaceName")
+      << m_instrumentWidget->getWorkspaceName().toStdString();
```

Comment:
These chained methods are forcing the client (in this case the `InstrumentViewWindow`) to know too much about the types involved. I would consider writing these as convenience methods on the `InstrumentWidget` such as `InstrumentWidget::getWorkspaceNameStdString()`